themes/default/content/blog/patterns-drift-detection/index.md

```diff
@@ -0,0 +1,302 @@
+---
+title: "Patterns for Drift Detection with Pulumi"
+
+# The date represents the post's publish date, and by default corresponds with
+# the date this file was generated. Posts with future dates are visible in development,
+# but excluded from production builds. Use the time and timezone-offset portions of
+# of this value to schedule posts for publishing later.
+date: 2022-06-16T09:26:46-05:00
+
+# Use the meta_desc property to provide a brief summary (one or two sentences)
+# of the content of the post, which is useful for targeting search results or social-media
+# previews. This field is required or the build will fail the linter test.
+# Max length is 160 characters.
+meta_desc: Curious about how to handle drift? Developer Advocate David shows a couple of patterns for detecting and reconciling drift with your Pulumi programs.
+
+# The meta_image appears in social-media previews and on the blog home page.
+# A placeholder image representing the recommended format, dimensions and aspect
+# ratio has been provided for you.
+meta_image: meta.png
+
+# At least one author is required. The values in this list correspond with the `id`
+# properties of the team member files at /data/team/team. Create a file for yourself
+# if you don't already have one.
+authors:
+    - david-flanagan
+
+# At least one tag is required. Lowercase, hyphen-delimited is recommended.
+tags:
+    - continuous-delivery
+
+# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md.
+# for additional details, and please remove these comments before submitting for review.
+---
+
+Let's face it, at some point someone is going to modify your carefully-crafted and automated infrastructure without updating your Pulumi program. These changes cause the desired state of our Pulumi program's to be inconsistent with the state of the world. These inconsistencies are often referred to as "drift". In this article, I want to cover a couple of patterns for detecting and reconciling this drift with your Pulumi programs.
+
+<!--more-->
+
+## Why is Drift Detection Important?
+
+Reconciling drift as quickly as possible is very important in the infrastructure as code (IaC) space because the longer drift is left unattended, the harder it becomes for automation to reconcile in a manner that can be considered safe for your organization, SLAs, and customers happiness. Imagine this as if it were your car. If a single tyre needs replaced, you might be OK for a while and not notice any problems driving. If you leave that unattended and another tyre runs flat, then you've got a slightly more serious problem. As these smaller problems accumulate you're applying a multiplier of risk to your overall infrastructure, especially during reconciliation. The more resources Pulumi needs to handle during this step, the more likely for some collateral damage.
```

Comment:
```suggestion
Reconciling drift as quickly as possible is very important in the infrastructure as code (IaC) space because the longer drift is left unattended, the harder it becomes for automation to reconcile in a manner that can be considered safe for your organization, SLAs, and customers' happiness. Imagine this as if it were your car. If a single tyre needs replaced, you might be okay for a while and not notice any problems driving. If you leave that unattended and another tyre runs flat, then you've got a slightly more serious problem. As these smaller problems accumulate, you're applying a multiplier of risk to your overall infrastructure, especially during reconciliation. The more resources Pulumi needs to handle during this step, the more likely for some collateral damage.
```